tests: mec15xxevb_assy6853: Enlarge the IDLE_STACK_SIZE from 256 to 512

hunterhu3000 opened the pull request:
The test case "samples/boards/mec15xxevb_assy6853/power_management" could
always pass until commit 5f60164 was merged in. This patch changes the sem
take timeout value from 50ms to forever of the log thread, it will make the
idle thread run longer and takes more stack memory. Analyzed the coredump
of the failed test case, it can be found that the test case stucks at
z_idle_stacks. The test case used default CONFIG_IDLE_STACK_SIZE = 256 in
kernel/Kconfig, now we increase the value to 512 in prj.conf for this test
only. With this change, the test case can pass with the commit of 5f60164.
Fixes #47987

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

nashif commented on subsys/logging/log_backend_uart.c:
```diff
@@ -136,6 +136,13 @@ static void log_backend_uart_init(struct log_backend const *const backend)
 
+static int is_ready(struct log_backend const *const backend)
+{
+	if (backend->cb)
```
curly braces here

hunterhu3000 replied:
Updated. But it seems there is another CI fail

nordic-krch commented on subsys/logging/log_core.c:
```diff
@@ -282,7 +282,11 @@ static uint32_t z_log_init(bool blocking, bool can_sleep)
+			/* if the backend is activated by the if code above,
+			 * set the mask
+			 */
+			if (log_backend_is_ready(backend)) {
```
why this change is needed?

hunterhu3000 replied:
Previously, log_backend_is_ready is checked only once, if the backend is not ready, it will try enable it. If it is ready, it will set the mask. When the backend is enabled, it will miss the setting of the mask, so I redo the check after that. 
If you think the below code looks better, I can replace with it 
···
                        if (log_backend_is_ready(backend) == 0) {
                                log_backend_enable(backend,
                                                   backend->cb->ctx,
                                                   CONFIG_LOG_MAX_LEVEL);
                                if (log_backend_is_ready(backend)) {
                                        mask |= BIT(i);
                                }
                        }
                        else {
                                         mask |= BIT(i);
                        }
···

nordic-krch commented:
See my comment https://github.com/zephyrproject-rtos/zephyr/issues/47987#issuecomment-1210191227
Imo this change is not needed and i'm not sure why it fixes #47987.

hunterhu3000 committed:
tests: mec15xxevb_assy6853: Enlarge the IDLE_STACK_SIZE from 256 to 512

The test case "samples/boards/mec15xxevb_assy6853/power_management" could
always pass until commit 5f60164 was merged in. This patch changes the sem
take timeout value from 50ms to forever of the log thread, it will make the
idle thread run longer and takes more stack memory. Analyzed the coredump
of the failed test case, it can be found that the test case stucks at
z_idle_stacks. The test case used default CONFIG_IDLE_STACK_SIZE = 256 in
kernel/Kconfig, now we increase the value to 512 in prj.conf for this test
only. With this change, the test case can pass with the commit of 5f60164.
Fixes #47987

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

MaureenHelm reviewed (changes requested):
Please explain the motivation for the change in the commit message

hunterhu3000 commented:
> Please explain the motivation for the change in the commit message
samples/boards/mec15xxevb_assy6853/power_management can pass until commit https://github.com/zephyrproject-rtos/zephyr/commit/5f60164a0fc36484b60004310320dfd6f3ef0525 merged in. This patch changes the sem take timeout of the log thread, and affect the run time of idle thread. Analyzed the coredump of the failed test case, it can be found that the program stucks at z_idle_stacks 
```
Remote debugging using 127.0.0.1:1234
0x0011ed78 in z_idle_stacks ()
(gdb) bt
#0  0x0011ed78 in z_idle_stacks ()
#1  0x000e2ab8 in arch_swap (key=<optimized out>) at /home/ztest/zephyrproject/zephyr/arch/arm/core/aarch32/swap.c:58
#2  0x0011ec40 in z_main_stack ()
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
(gdb) info thread
  Id   Target Id         Frame
* 1    Thread <main>     0x0011ed78 in z_idle_stacks ()
``` 
The CONFIG_IDLE_STACK_SIZE is set to 256 by default in the kernel/Kconfig. Change it to 512, and then the test case can pass.
The history can be seen in https://github.com/zephyrproject-rtos/zephyr/issues/47987

hunterhu3000 commented:
@MaureenHelm, the git commit message is updated as well. Thanks to review.
